cli/src/commands/git/push.rs

```diff
@@ -418,8 +463,72 @@ fn validate_commits_ready_to_push(
             }
             return Err(error);
         }
+        if commit.is_signed() != sign_settings.should_sign(commit.store_commit()) {
```

Comment:
nit: `!commit.is_signed() && sign_settings.should_sign(commit.store_commit())` is slightly easier to read to me, and makes it clearer that we don't worry about dropping signatures here (we can never have it `is_signed()` and not `should_sign()`)